### PR TITLE
Remove re-use of LockFileLibrary items

### PR DESCRIPTION
**Pull request by emgarten**
Remove re-use of LockFileLibrary items

Rename LocalNuspecCache to LocalPackageFileCache, this will now include the list of files from packages and the contents of the SHA512 file. Read the list of package files from the file cache instead of using the previous lock file to get the package file list.

* A lot of the changes here are due to renaming the cache.
* The actual logic change in this PR is in LockFileBuilder, it will no longer re-use the previous library.
* Performance with -Force looks the same or better based on the testing I did with Roslyn restore.
* Noop performance with parallel disabled went from 4.3s to 3.3s after due to caching sha512 files.

Fixes https://github.com/NuGet/Home/issues/5995
Fixes https://github.com/NuGet/Home/issues/6062

### CODE REVIEW

**Comment by nkolev92** [comment]
>             var previousLibraries = previousLockFile?.Libraries.ToDictionary(l => Tuple.Create(l.Name, l.Version));

Remove this? 

---
Refers to: src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs:49 in c324214. [](commit_id = c3242149154eaf6a64b4841ae42473fba74799f2, deletion_comment = False)

**Comment by emgarten** [comment]
@nkolev92 good catch, I'll remove that.

File: src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs [hunk 1]
```diff
@@ -52,10 +53,12 @@ public static IEnumerable<string> GetUniquePathsBasedOnOS(IEnumerable<string> pa
 
         /// <summary>
         /// Replace all back slashes with forward slashes.
+        /// If the path does not contain a back slash
+        /// the original string is returned.
         /// </summary>
         public static string GetPathWithForwardSlashes(string path)
         {
-            if (path != null && path.IndexOf('\\') >= -1)
+            if (path != null && path.IndexOf('\\') > -1)
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs:61]: Well glad this fuck-up was harmless lol :)
💯

**Review by nkolev92 — APPROVED** [review]
LGTM
🏆

**Pull request merged**